### DISCUSSION TITLE
Bump deprecated/unsupported ranges

### DISCUSSION THREAD

**Discussion by rarkins** [renovatebot/renovate#21664] [Suggest an Idea]
Best example is Node.js:

If a user has an engines like this:

```
"engines": {
  "node": ">=4.2.0"
}
```

It should be upgraded to this:
```
"engines": {
  "node": ">=6.9.0"
}
```

i.e. to the minimum supported (LTS) version. Or more specifically, "Bump ranges so that their minimum value is the lowest non-deprecated version".

This is related to #1751

**Comment by daniellockyer** [comment]
Hey @rarkins, is this still on the roadmap? We use Renovate at Ghost and I believe it's not bumping our `package.json` versions because of this issue. Thanks!

**Comment by rarkins** [comment]
@daniellockyer could you give an example, so that I can establish if it's the same cause as the subject of this issue?

**Comment by daniellockyer** [comment]
@rarkins Sure! That'd be great.

* We have [Ghost](https://github.com/tryghost/ghost) with the [`lts` supportPolicy](https://github.com/TryGhost/Ghost/blob/master/renovate.json#L11) and `^8.16.0 || ^10.13.0 || ^12.10.0` in `package.json` but it doesn't seem to get updated to remove `8.x`.
* We have tools like [gscan](https://github.com/TryGhost/gscan) also with the [`lts` supportPolicy](https://github.com/TryGhost/gscan/blob/master/renovate.json#L10) but the [incoming PR](https://github.com/TryGhost/gscan/pull/283/files) from Renovate doesn't [actually update the range](https://github.com/TryGhost/gscan/blob/master/package.json#L17) we have (`>=8`).

This is potentially two different types of declarations it'd have to deal with, but Renovate doesn't seem to deal with either. Let me know if you need more info!

**Comment by rarkins** [comment]
Thanks, these are related/overlapping issues.
